syntax = "proto2";
package antalk.auth;
import "common.proto";
option java_package = "com.antalk.pb";
//option optimize_for = LITE_RUNTIME;

option cc_generic_services = true;

message LoginReq {
    optional string uid = 1;
	optional string user_name = 2;
	optional string password = 3;
	optional antalk.common.UserStatType online_status = 4;
	optional antalk.common.ClientType client_type = 5;
	optional string client_version = 6;
}

message LoginResp {
	optional uint32 server_time = 1;
	optional antalk.common.ResultType result_code = 2;
	optional string result_string = 3;
	optional antalk.common.UserStatType online_status = 4;
	optional antalk.common.UserInfo user_info = 5;
}

message LogoutReq {
	optional string uid = 1;
	optional int64 time_stamp = 2;
}

message LogoutResp {
}

service LoginService {
      rpc Login(LoginReq) returns (LoginResp);
      rpc Logout(LogoutReq) returns (LogoutResp);
};
